src/mono/cmake/configure.cmake

```diff
@@ -94,6 +94,21 @@ endif()
 
 check_include_files("sys/types.h;sys/user.h" HAVE_SYS_USER_H)
 
+check_c_source_compiles(
+  "
+  #include <zlib.h>
+  int main(void)
+  {
+    #if defined(ZLIB_VERNUM) && (ZLIB_VERNUM >= 0x1230)
+    #else
+    #error No good zlib found
+    #endif
+    return 0;
```

Comment:
This is copied from

https://github.com/mono/mono/blob/a791978a6e4312698137be451c648b650a8c8663/configure.ac#L728-L732

but I'm not sure if that's necessary.

Reply:
We can include (`include(${CLR_SRC_NATIVE_DIR}/libs/System.IO.Compression.Native/extra_libs.cmake)`), call (`set(Z_LIBS)\nappend_extra_compression_libs(Z_LIBS)`) this macro: https://github.com/dotnet/runtime/blob/60012a1092e6cb0f4fe6fc9d9ffd9907826a8355/src/native/libs/System.IO.Compression.Native/extra_libs.cmake#L2  and skip the introspection check. The difference is that in libs, _when_ we require zlib package on the build machine, we fail the build if it was not found. In mono, we treat it as optional dependency. It would be nice to streamline the requirements and cleanup `!build_from_source && !zlib_found` fallback code.

Reply:
@am11 Yea I think this probably needs a bit more work for net7.  I think the logic in Mono is kind of backwards: we probe for zlib (or in some cases statically link the in-tree copy) and based on that we enable or disable certain features in the runtime (Embedded PDB support; compressed log profiler output).  I think that's backwards.  We should have feature options and those should drive the zlib dependency decisions.

I'll rework it and include the cmake fragment from libs.

Reply:
A note on another difference between mono and rest of the native partitions in repo: all native partitions use `eng/native/cofigureplatform.cmake`, which defines `CLR_CMAKE_{HOST, TARGET}_*` properties. In mono, we only use configure*.cmake in `src/mono/dlls/*`, for the remaining areas, we define a separate set of platform properties. In future, if we change mono to also use common properties, it will be possible to share more configuration code without reinterpreting the semantics.

As it stands, `append_extra_compression_libs` has these differences:

|  rest of the repo | mono |
| --- | --- |
| `CLR_CMAKE_TARGET_BROWSER` | `HOST_BROWSER` |
| `CLR_CMAKE_TARGET_ANDROID` | `HOST_ANDROID` |
| `CLR_CMAKE_TARGET_SUNOS` | `HOST_SOLARIS` |

Reply:
Oh that's annoying. 

Do you think it makes sense to modify `append_extra_compression_libs` to also check for the mono names, @am11?

(In the long term we should use the `CLR_CMAKE_` names everywhere, and change mono's `HOST_ABC` macros to `CLR_CMAKE_TARGET_ABC` and mono's `TARGET_XYZ` macros to something like `CODEGEN_TARGET_XYZ` - mono uses target to mean the platform for which the JIT or AOT compiler is generating code)

Reply:
> Do you think it makes sense to modify `append_extra_compression_libs` to also check for the mono names, @am11?

Yup, that will work. `if (CLR_CMAKE_TARGET_BROWSER OR HOST_BROWSER)` etc.

> mono uses target to mean the platform for which the JIT or AOT compiler is generating code

I understand this difference and hope we don't carry it in the future by using same semantics everywhere. :slightly_smiling_face:
On a closer look, HOST and TARGET have different meaning in mono vs. other repo partitions. Earlier, Steve streamlined the difference in for other partitions in https://github.com/dotnet/runtime/pull/1974. It is mostly how VS / Windows use the terms. Mono was originally using Unix-y autoconf, which has a bit different meanings; `Host` is what VS calls `Target` and `Build` is what VS refers to as `Host`. `Target` in autoconf (and hence mono) is a special case, where compiler itself is built on a different architecture than `Host` and `Build`, and `TARGET` is used to indicate the machine for which code is generated:
  
* https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Specifying-Target-Triplets.html
* https://stackoverflow.com/a/15901574

Reply:
> I understand this difference and hope we don't carry it in the future by using same semantics everywhere. 

yea I agree.  (it's my long (loooong) term hope that mono's AOT compiler can be made multitargeting so the autoconf-sense `target` will go away entirely as a build-time idea and we can pick the AOT output triple when we invoke the AOT compiler)